README.md

```diff
@@ -4,130 +4,180 @@
 
 # Swift for TensorFlow
 
-Welcome to the Swift for TensorFlow development community! For discussions, join the [swift@tensorflow.org mailing list](https://groups.google.com/a/tensorflow.org/d/forum/swift).
+> S4TF: No barriers.
+
+Swift for TensorFlow (aka S4TF) is a next-generation machine learning framework,
+incorporating the latest research across: machine learning, compilers, 
+differential programming, systems design, and beyond. This project is
+approaching version _0.2_; it is neither feature complete nor production-ready.
+But it is ready for _pioneers_ to try it for your own projects, give us
+feedback, and help shape the future!
+
+The S4TF project is currently focusing on 2 kinds of users:
+
+ 1. **Advanced ML Researchers** who are limited by current ML frameworks.
+    S4TF's advantages include a seamless integration with a modern general-purpose
+    language, allowing for more dynamic and sophisticated models. Fast
+    abstractions can be developed "in user-space" (as opposed to in C/C++
+    aka "framework-space"), resulting in modular APIs that can be easily
+    customized.
+
+ 2. **ML Learners** who are just getting started with machine learning. Thanks
+    to Swift's support for quality tooling (e.g. context-aware autocomplete),
+    S4ML can be one of the most productive ways to get started learning the
+    fundamentals of machine learning.
+
+## Getting started
+
+ - **Colab**: The fastest way to get started is to try out S4TF right in your
+   browser. Just open up our [getting started
+   notebook](https://colab.research.google.com/github/tensorflow/swift-tutorials/blob/master/iris/swift_tensorflow_tutorial.ipynb) (or start from a
+   [blank notebook](https://colab.research.google.com/github/tensorflow/swift/blob/master/notebooks/blank_swift.ipynb))!
+   Read more in our [usage guide](Usage.md).
+
+ - **Install locally**: you can [download a pre-built Swift for TensorFlow
+   package](Installation.md). After installation, you can follow these
+   [step-by-step instructions](Usage.md) to build and execute a S4TF program
+   on your computer.
+
+ - **Compile from source**: If you'd like to customize S4TF or even contribute
+   back, follow our [instructions on building the S4TF compiler from source,
+   from in the compiler's README](https://github.com/apple/swift/tree/tensorflow).
```

Comment:
It's better to avoid links that are too long. In this case, the sentence is just too long and the second half feels redundant.

Just "instructions on building the compiler from source" would be good.